transport/ssh/ssh.go

```diff
@@ -46,6 +46,16 @@ func Dial(ctx context.Context, network, addr string, config *ssh.ClientConfig) (
 	return newTransport(client, true)
 }
 
+// DialWithConn is same as Dial but creates the transport on top of input net.Conn
+func DialWithConn(addr string, config *ssh.ClientConfig, conn net.Conn) (*Transport, error) {
```

Comment:
I don't want to add bunch of "Dial*" type functions like I did in `github.com/Juniper/netconf`  `ssh.NewTransport` should fit everything that that `Dial` cannot.  This logic can just but stuffed into  the session creation in callhome.go.

If for some reason it becomes very popular that sessions (outside of callhome) need to contruct already from a net.Con we can look at potentially adding this back, but for now I want the API surface to be small.

Easier to add functions in a minor version then it is to remove them in a major verion.